oaipmh/requests/data_queries.py

```diff
@@ -0,0 +1,143 @@
+    query_data: Dict[OAIParams, str]={OAIParams.VERB:OAIVerbs.GET_RECORD}
+
+    # get parameters
+    expected_params={OAIParams.ID, OAIParams.META_PREFIX, OAIParams.VERB}
+    if set(params.keys()) != expected_params:
+        raise OAIBadArgument(f"Parameters provided did not match expected. Expected: {', '.join(str(param) for param in expected_params)}")
+    
+    identifier_str=params[OAIParams.ID]
+    arxiv_id=process_identifier(identifier_str)
+    query_data[OAIParams.ID]=identifier_str
+
+    meta_type_str=params[OAIParams.META_PREFIX]
+    if meta_type_str not in SUPPORTED_METADATA_FORMATS:
+        raise OAIBadFormat(reason="Did not recognize requested format", query_params=query_data)
+    meta_type=SUPPORTED_METADATA_FORMATS[meta_type_str]
+
+    #TODO paramters done, do rest of function
+
+    return "<a>b</a>", 200, {}
+
+def list_records(params: Dict[str, str]) -> Response:
+    """used to harvest records from a repository with support for selective harvesting"""
+    return _list_data(params, False)
+
+def list_identifiers(params: Dict[str, str]) -> Response:
+    """retrieves headers of all records matching certain parameters"""
+    return _list_data(params, True)
+
+
+def _list_data(params: Dict[str, str], just_ids: bool)-> Response:
+    """runs both list queries. just_ids true for list identifiers, false for list records"""
+    query_data: Dict[OAIParams, str]={OAIParams.VERB:OAIVerbs.LIST_IDS}
+
+    #parameter processing
+    given_params=set(params.keys())
+    if OAIParams.RES_TOKEN in given_params: #using resumption token
+        if given_params != {OAIParams.RES_TOKEN, OAIParams.VERB}: #resumption token is exclusive
+            raise OAIBadArgument(f"No other paramters allowed with {OAIParams.RES_TOKEN}")
+        token=params[OAIParams.RES_TOKEN]
+        #TODO token processing and validation
+
+    else: #using request parameters
+        #correct parameters present
+        if OAIParams.META_PREFIX not in given_params:
+            raise OAIBadArgument(f"{OAIParams.META_PREFIX} required.")
+        allowed_params={OAIParams.VERB,OAIParams.META_PREFIX, OAIParams.FROM, OAIParams.UNTIL, OAIParams.SET }
+        if given_params-allowed_params: #no extra keys allowed
+            raise OAIBadArgument(f"Unallowed parameter. Allowed parameters: {', '.join(str(param) for param in allowed_params)}")
+
+        #metadata
+        meta_type_str=params[OAIParams.META_PREFIX]
+        if meta_type_str not in SUPPORTED_METADATA_FORMATS:
+            raise OAIBadFormat(reason="Did not recognize requested format", query_params=query_data)
+        meta_type=SUPPORTED_METADATA_FORMATS[meta_type_str]
+        query_data[OAIParams.META_PREFIX]=meta_type_str
+
+        #dates
+        from_str=params.get(OAIParams.FROM)
+        if from_str:
+            try:
+                if not re.fullmatch(DATE_REGEX, from_str):
+                    raise ValueError
```

Comment:
This is redundant since strptime will throw a ValueError

Reply:
strptime is a little too permissive and accepts things like 3-2-2024 which the specification says must be 03-02-2024